3main

```diff
@@ -524,42 +479,42 @@ then
+      eval gpu_clks_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\$((_CORE_OVERCLOCK + __CORE_OVERCLOCK_/' -e 's/$/))/' | paste -s -d, -)]"
+      eval mem_clks_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\$((_MEMORY_OVERCLOCK + MEMORY_OVERCLOCK_/' -e 's/$/))/' | paste -s -d, -)]"
     elif [[ $OVERCLOCK_MODE == GPU_SPECIFIC ]]
     then
-      eval gpu_clks_$a="[$(seq 0 $(($GPUS-1)) | sed -e 's/^/\${__CORE_OVERCLOCK_/' -e 's/$/}/' | paste -s -d, -)]"
-      eval mem_clks_$a="[$(seq 0 $(($GPUS-1)) | sed -e 's/^/\${MEMORY_OVERCLOCK_/' -e 's/$/}/' | paste -s -d, -)]"
+      eval gpu_clks_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\${__CORE_OVERCLOCK_/' -e 's/$/}/' | paste -s -d, -)]"
+      eval mem_clks_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\${MEMORY_OVERCLOCK_/' -e 's/$/}/' | paste -s -d, -)]"
     elif [[ $OVERCLOCK_MODE == ALGO_SPECIFIC ]]
     then
       eval gpu_clks_$a=$((${a}_CORE_OVERCLOCK))
       eval mem_clks_$a=$((${a}_MEMORY_OVERCLOCK))
     elif [[ $OVERCLOCK_MODE == ALGO_SPECIFIC_with_GPU_OFFSET ]]
     then
-      eval gpu_clks_$a="[$(seq 0 $(($GPUS-1)) | sed -e 's/^/\$((${a}_CORE_OVERCLOCK + __CORE_OVERCLOCK_/' -e 's/$/))/' | paste -s -d, -)]"
-      eval mem_clks_$a="[$(seq 0 $(($GPUS-1)) | sed -e 's/^/\$((${a}_MEMORY_OVERCLOCK + MEMORY_OVERCLOCK_/' -e 's/$/))/' | paste -s -d, -)]"
+      eval gpu_clks_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\$((${a}_CORE_OVERCLOCK + __CORE_OVERCLOCK_/' -e 's/$/))/' | paste -s -d, -)]"
+      eval mem_clks_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\$((${a}_MEMORY_OVERCLOCK + MEMORY_OVERCLOCK_/' -e 's/$/))/' | paste -s -d, -)]"
     fi
 
     if [[ $POWERLIMIT_MODE == GLOBAL ]]
     then
       eval pwr_lim_$a=$_POWERLIMIT_WATTS
     elif [[ $POWERLIMIT_MODE == GLOBAL_with_GPU_OFFSET ]]
     then
-      eval pwr_lim_$a="[$(seq 0 $(($GPUS-1)) | sed -e 's/^/\$((_POWERLIMIT_WATTS+INDIVIDUAL_POWERLIMIT_/' -e 's/$/))/' | paste -s -d, -)]"
+      eval pwr_lim_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\$((_POWERLIMIT_WATTS+INDIVIDUAL_POWERLIMIT_/' -e 's/$/))/' | paste -s -d, -)]"
     elif [[ $POWERLIMIT_MODE == GPU_SPECIFIC ]]
     then
-      eval pwr_lim_$a="[$(seq 0 $(($GPUS-1)) | sed -e 's/^/\${INDIVIDUAL_POWERLIMIT_/' -e 's/$/}/' | paste -s -d, -)]"
+      eval pwr_lim_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\${INDIVIDUAL_POWERLIMIT_/' -e 's/$/}/' | paste -s -d, -)]"
     elif [[ $POWERLIMIT_MODE == ALGO_SPECIFIC ]]
     then
       eval pwr_lim_$a=$((${a}_POWERLIMIT_WATTS))
     elif [[ $POWERLIMIT_MODE == ALGO_SPECIFIC_with_GPU_OFFSET ]]
     then
-      eval pwr_lim_$a="[$(seq 0 $(($GPUS-1)) | sed -e 's/^/\$((${a}_POWERLIMIT_WATTS+INDIVIDUAL_POWERLIMIT_/' -e 's/$/))/' | paste -s -d, -)]"
+      eval pwr_lim_$a="[$(seq 0 $((GPUS-1)) | sed -e 's/^/\$((${a}_POWERLIMIT_WATTS+INDIVIDUAL_POWERLIMIT_/' -e 's/$/))/' | paste -s -d, -)]"
     fi
   done
 fi
 
-if [[ $COIN == SALFTER_NICEHASH_PROFIT_SWITCHING && $AUTO_START_MINER == YES ]]
+if [[ $AUTO_SWITCH == SALFTER_NICEHASH_SWITCHING ]]
 then
   HOSTEDIT="YES"
```

Comment:
Code clean up:
```
if [[ $AUTO_SWITCH == SALFTER_NICEHASH_SWITCHING ]]
then
  if ! grep -q "api.nicehash.com" /etc/hosts;
  then
    sudo -- sh -c -e "echo '' >> /etc/hosts";
    sudo -- sh -c -e "echo '104.20.158.21 api.nicehash.com' >> /etc/hosts";
    echo UPDATED HOSTS FILE WITH IPv6 FIX
    echo ""
    echo ""
  fi
```

Reply:
It can be shorter :

`if ! grep -q "api.nicehash.com" /etc/hosts && [[ $AUTO_SWITCH == SALFTER_NICEHASH_SWITCHING ]]  ; then`

Reply:
It depends how bash expressions are evaluated. if it supports lazy evaluation than your last expression would be ok with the two expressions in reverse order. Otherwise if evaluation is not lazy the grep test must be kept aside as you have done two comments above, otherwise we have an useless read access to /etc/hosts in case AUTO_SWITCH is not *_NICEHASH_*

Reply:
Yup, reverse is a better idea